UI changes according to support vertical flowchart and scrolling

utsavgarg opened the pull request:
This commit adds changes in accordance to #14 and issue #16. 
Changes as below:
- bigger size of boxes at the start
- flowchart extends vertically
- canvas scrolls using scroll wheel
- panning works as before by dragging canvas
- double click zooms and ctrl + double click shrinks
- '[' key zooms and ']' key shrinks
- tested for googleNet, resNet, alexNet, vgg and cascaded fcn
- made all buttons equal sized and added border radius
- also added tooltips on hover

virajprabhu commented on ide/static/css/style.css:
```diff
@@ -73,6 +73,22 @@ html,body,#app,.app{
+.rhTooltip{
+  padding: 0px;
+}
+
```
Remove extra line

virajprabhu commented on ide/static/js/layer.js:
```diff
@@ -25,6 +25,10 @@ class Layer extends React.Component {
+        onMouseEnter={(event) => this.props.hover(event, this.props.id)}
+        data-tip
+        data-for='getContent'
+
```
Remove newline

virajprabhu commented on ide/static/js/netLayout_vertical.js:
```diff
@@ -0,0 +1,120 @@
+// but the UI may not be clean.
+
+export default function(net){
+// map[x] = [y1, y2, y3]
```
Remove unused code

virajprabhu commented on ide/static/js/netLayout_vertical.js:
```diff
@@ -0,0 +1,120 @@
+  }
+  let positionsX = map[preferredPosition[1]];
+  if (positionsX.indexOf(preferredPosition[0]) != -1) {
+    let temp = preferredPosition[0], i=2;
```
Add space around assignment operator.

virajprabhu commented on ide/static/js/panZoom.js:
```diff
@@ -105,7 +112,7 @@ export default function() {
 
   if ("onmousewheel" in document) { panZoom.onmousewheel = mousewheel; }
```
We should allow pinch to zoom out.

virajprabhu replied:
This works fine now.

virajprabhu commented on ide/static/js/tooltip.js:
```diff
@@ -0,0 +1,67 @@
+      const params = [];
+      const props = [];
+      const layer = this.props.net[this.props.hoveredLayer];
+      //console.log(layer);
```
Remove logs

utsavgarg commented:
@virajprabhu I have made the initial boxes slightly smaller, changed the positioning to adapt to the screen size and done code clean up. I also added functionality for pinch to zoom-in/out but do not have a way to check if it works.

virajprabhu commented on ide/static/js/content.js:
```diff
@@ -242,8 +260,8 @@ class Content extends React.Component {
-        top: `${250 + 50 * positions[layerId][1]}px`,
-        left: `${20 + 180 * positions[layerId][0]}px`,
+        top: `${height + 65 * positions[layerId][1]}px`,
+        left: `${width + 80 * positions[layerId][0]}px`,
```
I'm not sure I still follow. Why is the offset hardcoded? Still seems a little off.
![screen shot 2017-05-13 at 9 32 56 am 3](https://cloud.githubusercontent.com/assets/8670301/26025891/3f383366-37bf-11e7-8ffe-9df30dd683f4.png)

utsavgarg replied:
The height and width are now based on the browser innerHeight and width, 65 and 80 are for relative spacing between nodes. Not quite sure why its still not centring for other resolutions.

virajprabhu commented on ide/static/js/panZoom.js:
```diff
@@ -50,10 +56,15 @@ export default function() {
   };
 
   window.onmouseup = function() {
-  //panZoom.onmouseup = function() {
```
Delete line.

virajprabhu commented on ide/static/js/panZoom.js:
```diff
@@ -68,13 +79,14 @@ export default function() {
       canvas.y += e.pageY - previousMousePosition.y;
       canvas.updateContainerPosition();
       previousMousePosition = { x: e.pageX, y: e.pageY };
-      //instance.repaintEverything();
```
Delete.

virajprabhu commented on ide/static/js/panZoom.js:
```diff
@@ -85,15 +97,14 @@ export default function() {
   }
 
-  var mousewheel, lastMouseWheelEventTime = Date.now();
+  /*var mousewheel, lastMouseWheelEventTime = Date.now();
```
Delete.

virajprabhu commented:
Changes incorporated in #67.
